forem/forem
# /profile/settings page rendering (Ridhwana/Michael)

_The original PR with comments was https://github.com/forem/forem/pull/10272, but Ridwhana had some issues with the checkout so we started a new branch._

## What type of PR is this? (check all applicable)


## Description

The profile settings page now renders dynamically from the `ProfileFieldGroup` model and `ProfileField` model. 
Updates to `ProfilesController` are sent to `/profiles/:id`. We re-use the `Profiles::Update` service to split the user and the generalized profile fields to update the relevant models. 

Need to:

- [x] Fix the checkboxes and styling on the ui
- [x] Fix the checkboxes, it's currently not working 
- [x] Fix the upload of the profile image (move the validation etc. as well)
- [x] Show feedback on the page when success or errors occurs
- [x] Need to check the success of both the profile updates and the user updates

## Related Tickets & Documents

Part of #6994

## QA Instructions, Screenshots, Recordings

1. Go to `/settings/profile`
2. Edit various attributes and observe that the changes are reflected on the page as expected.
3. Optional: use the Rails console to verify that the attributes are synced between users and profiles.

## Added tests?


## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
